include/cantera/zeroD/FlowReactor.h

```diff
@@ -6,55 +6,131 @@
 #ifndef CT_FLOWREACTOR_H
 #define CT_FLOWREACTOR_H
 
-#include "Reactor.h"
+#include "IdealGasReactor.h"
 
 namespace Cantera
 {
 
-//! Adiabatic flow in a constant-area duct.
-class FlowReactor : public Reactor
+//! Adiabatic flow in a constant-area duct with homogeneous and heterogeneous reactions
+class FlowReactor : public IdealGasReactor
 {
 public:
     FlowReactor() = default;
 
-    virtual std::string type() const {
+    string type() const override {
         return "FlowReactor";
     }
 
-    virtual void getState(doublereal* y);
+    bool isOde() const override {
+        return false;
+    }
 
-    virtual void initialize(doublereal t0 = 0.0);
-    virtual void eval(double t, double* LHS, double* RHS);
-    virtual void updateState(doublereal* y);
+    //! Not implemented; FlowReactor implements getStateDAE() instead.
+    void getState(double* y) override {
+        throw NotImplementedError("FlowReactor::getState");
+    }
 
-    void setMassFlowRate(double mdot);
+    void getStateDae(double* y, double* ydot) override;
+    void initialize(double t0=0.0) override;
+    void syncState() override;
+    void updateState(double* y) override;
 
-    void setTimeConstant(doublereal tau) {
-        m_fctr = 1.0/tau;
+    //! Not implemented; FlowReactor implements evalDae() instead.
+    void eval(double t, double* LHS, double* RHS) override {
+        throw NotImplementedError("FlowReactor::eval");
     }
 
+    void evalDae(double t, double* y, double* ydot, double* residual) override;
+
+    void getConstraints(double* constraints) override;
+
+    //! Set the mass flow rate through the reactor [kg/s]
+    void setMassFlowRate(double mdot);
+
+    //! The current gas speed in the reactor [m/s]
     double speed() const {
-        return m_speed;
+        return m_u;
+    }
+
+    //! The cross-sectional area of the reactor [m^2]
+    double area() const {
+        return m_area;
+    }
+
+    //! Sets the area of the reactor [m^2]
+    void setArea(double area);
+
+    //! The ratio of the reactor's surface area to volume ratio [m^-1]
+    //! @note If the surface area to volume ratio is unspecified by the user,
+    //!       this will be calculated assuming the reactor is a cylinder.
+    double surfaceAreaToVolumeRatio() const;
+
+    //! Set the reactor's surface area to volume ratio [m^-1]
+    void setSurfaceAreaToVolumeRatio(double sa_to_vol) {
+        m_sa_to_vol = sa_to_vol;
+    }
+
+    //! Set the steady state tolerances used to determine the initial state for
+    //! surface coverages
+    void setInletSurfaceAtol(double atol) {
+        m_ss_atol = atol;
     }
-    double distance() const {
```

Comment:
Would you reconsider the removal of `distance`? In some contexts (examples!), using time as a placeholder for distance is somewhat confusing, and also not universally applicable.